### PR TITLE
[14.0][ADD] hr_timesheet_activity_begin_end_readonly_duration

### PR DESCRIPTION
**Pull request by ChristianSantamaria**
Change the readonly attribute of field unit_amount to true, to force the Begin Hour and End Hour to be added

### CODE REVIEW

File: hr_timesheet_activity_begin_end_readonly_duration/models/analytic_account.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import fields, models, api
+from datetime import timedelta
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = "account.analytic.line"
+
+    unit_amount = fields.Float(string="Quantity", default=0.0, readonly=True)
```
Comment (dalonsod) [inline hr_timesheet_activity_begin_end_readonly_duration/models/analytic_account.py]: If only `readonly` has changed, only this code addition should be needed:
```suggestion
    unit_amount = fields.Float(readonly=True)
```

File: hr_timesheet_activity_begin_end_readonly_duration/models/analytic_account.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    @api.constrains("time_start", "time_stop")
+    def _check_time_start_stop(self):
+        for line in self:
+            start = timedelta(hours=self.time_start)
+            stop = timedelta(hours=self.time_stop)
+            line.unit_amount = (stop - start).seconds / 3600
```
Comment (dalonsod) [inline hr_timesheet_activity_begin_end_readonly_duration/models/analytic_account.py]: Why is this code needed? If it's finally required, better try this (code reuse, from https://github.com/OCA/timesheet/blob/14.0/hr_timesheet_activity_begin_end/models/account_analytic_line.py#L72):
```suggestion
            line.onchange_hours_start_stop()
```

File: hr_timesheet_activity_begin_end_readonly_duration/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# © 2023 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+{
+    "name": "Timesheet Activities - Begin/End Hours - Readonly Duration",
+    "summary": """
+        Change the readonly attribute of field unit_amount to true
+        to force the Begin Hour and End Hour to be added
+    """,
+    "author": "Solvos",
+    "license": "LGPL-3",
```
Comment (dalonsod) [inline hr_timesheet_activity_begin_end_readonly_duration/__manifest__.py]: This module depends on an OCA addon, so we should this license instead
```suggestion
    "license": "AGPL-3",
```

**Review by dalonsod — CHANGES_REQUESTED** [review]
Only code review, see comments

**Comment by ChristianSantamaria** [comment]
Done 👍

**Comment by dalonsod** [comment]
OK, I'll test it ASAP!

**Comment by dalonsod** [comment]
Locally tested, 👍 

There's only a minor problem in form view edition (you can reach it using kanban view): `hr_timesheet_activity_begin_end` doesn't provide `time_start` and `time_stop` fields here. Then, we're unable to set timesheet hours.

So, I've temporary added them (this sould be a later contribution to OCA IMO).

@ChristianSantamaria could you install it in ArvTest environment?

**Comment by ChristianSantamaria** [comment]
Instaled and tested in test enviroment, LGTM 👍

**Comment by dalonsod** [comment]
Great!
I think the two separate commits should be kept in this situation, so no squash needed. You can follow the OCA proposed patch here: OCA/timesheet#577

**Pull request merged**